### PR TITLE
[Testcase Refactoring] Label test_fsdp_clip_grad_norm and decouple from hard-coded device list

### PR DESCRIPTION
**Pull request by twthighd**
## Summary

This PR labels `TestClipGradNorm` as `HardwareClassification.ACCELERATOR`, switches its `instantiate_device_type_tests` from a hard-coded `only_for=("cuda","hpu","xpu")` whitelist to `except_for=("cpu",)`, and completes the device-parameter migration in the test bodies — using `self.device_type` (the per-variant class attribute) for FSDP `device_id`, DDP `device_ids`, and all model/tensor creation including `get_input(...)` so each rank resolves to its own device (review feedback).

## Changes

- **`test/distributed/fsdp/test_fsdp_clip_grad_norm.py`**:
  - Added `hw_classification = HardwareClassification.ACCELERATOR` to `TestClipGradNorm`, plus the `HardwareClassification` import. All five `test_*` methods require an accelerator (`@skip_if_lt_x_gpu(2)`, `device` param) and exercise FSDP `clip_grad_norm_()` on the device-tensor path, so `ACCELERATOR` is the correct classification.
  - Replaced `devices = ("cuda", "hpu", "xpu")` + `only_for=devices` with `except_for=("cpu",)` in `instantiate_device_type_tests`, so any accelerator registered via `torch.accelerator` (including PrivateUse1-based out-of-tree backends) is admitted instead of a hard-coded whitelist. This aligns with the TEST_LINTER direction (PR #190173): `ACCELERATOR` classes must NOT use `only_for` (use `except_for` blacklist).
  - Completed the device-parameter migration in the test bodies: deleted the module-level `device_type = torch.device(get_devtype())` and the `get_devtype` import, and replaced every in-body reference (`device_type.type`, `self.device_type`, `device_ids=[device_type]`, `get_input(device)`) with `self.device_type` — the per-variant class attribute whose value equals the injected device's type-only string, matching the convention of the sibling FSDP test files (review feedback). FSDP `device_id`, DDP `device_ids`, `nn.Linear`, `torch.randn`, `.to()`, and `get_input()` all use it; this fixes the latent mismatch where the module-level constant was a single value that did not change per device variant.

## Motivation

`TestClipGradNorm` had no `hw_classification`, so it was silently dropped under `--hw-classification ACCELERATOR` filtering. Its `instantiate_device_type_tests` used `only_for=("cuda","hpu","xpu")`, excluding PrivateUse1-based out-of-tree accelerators even when available. Additionally, the test bodies read a module-level `device_type` constant — a single value that does not change across per-device variants — which under `instantiate_device_type_tests` would use the wrong device on non-default variants. Completing the migration to `self.device_type` (the per-variant class attribute, preserving per-rank device resolution) fixes this mismatch and aligns the file with the other FSDP test refactors.

## Test Plan

- Verified the FSDP test infra routes to the accelerator backend (not a gloo/CPU false pass):

```
python -c "from torch.testing._internal.common_fsdp import DEVICE_TYPE, DISTRIBUTED_BACKEND, DEVICE_COUNT; print(DEVICE_TYPE, DISTRIBUTED_BACKEND, DEVICE_COUNT)"
# <device> <backend> <count>  (e.g. an accelerator backend, not cpu gloo 1)
```

- End-to-end on an 8-card accelerator backend (with the `common_fsdp.py` accelerator branch from #187650 applied):

```
python test/distributed/fsdp/test_fsdp_clip_grad_norm.py --hw-classification ACCELERATOR
# 5/5 pass; FSDP warnings confirm per-rank device resolution
# (rank 0 -> device 0, rank 3 -> device 3, ...) — true multi-card
```

- `lintrunner` clean; `review-test-refactoring` skill review: no findings.

## Notes

- `TestClipGradNorm` retains `@skip_if_lt_x_gpu(2)` on all five methods unchanged. Making `skip_if_lt_x_gpu` recognize out-of-tree accelerators is a separate framework-level change to `common_distributed.py`, tracked in pytorch/pytorch#187650; this PR does not modify that helper.
- This is part of the test case refactoring initiative tracked in #185590.

### CODE REVIEW

**Comment by twthighd** [comment]
@wjlFlyer @pjyFight @FuDdd @fffrog @lyriexs666 @jishuangfeng @JamesD18 @JiasenTian

**Review by wjlFlyer — APPROVED** [review]

File: test/distributed/fsdp/test_fsdp_clip_grad_norm.py [hunk 1]
```diff
@@ -59,11 +62,11 @@ def __init__(self) -> None:
             def forward(self, x: torch.Tensor) -> torch.Tensor:
                 return self.lin2(self.lin1(x))
 
-        model = Model().to(device_type.type)
+        model = Model().to(torch.device(device).type)
```
Comment (orangeH25) [inline test/distributed/fsdp/test_fsdp_clip_grad_norm.py]: Could we use `self.device_type` directly here and in the other similar places, instead of `torch.device(device).type`?

**Review by orangeH25 — COMMENTED** [review]
Hi @twthighd , I left a comments. Please take a look when you have a chance.

File: test/distributed/fsdp/test_fsdp_clip_grad_norm.py (hunk 1 above)
Comment (twthighd) [reply test/distributed/fsdp/test_fsdp_clip_grad_norm.py]: Done — replaced torch.device(device).type with self.device_type at all 12 occurrences in the file (not just line 65), matching the convention of the sibling FSDP test files. Verified end-to-end on an 8-card accelerator backend: 5/5 pass.Thanks for the review!

File: test/distributed/fsdp/test_fsdp_clip_grad_norm.py [hunk 2]
```diff
@@ -340,12 +342,13 @@ def _test_no_gradients(self, device, use_orig_params: bool):
         ):
             total_norm = fsdp_module.clip_grad_norm_(1)
         self.assertEqual(total_norm.dtype, torch.float32)
-        self.assertEqual(total_norm, torch.tensor(0.0, device=self.device_type))
+        expected = torch.tensor(0.0, device=self.device_type)
+        self.assertEqual(total_norm, expected)
```
Comment (orangeH25) [inline test/distributed/fsdp/test_fsdp_clip_grad_norm.py]: These two lines can be reverted since there are no actual changes here.

Comment (twthighd) [reply test/distributed/fsdp/test_fsdp_clip_grad_norm.py]: Done — reverted to the original single-line self.assertEqual(total_norm, torch.tensor(0.0, device=self.device_type)); the two-line split was a leftover from an intermediate iteration where the longer form exceeded the line-width limit. That hunk no longer appears in the diff.

Thanks for the review!

**Review by orangeH25 — APPROVED** [review]
